core/src/main/java/io/undertow/client/http/HttpClientConnection.java

```diff
@@ -110,25 +110,24 @@ public void handleEvent(StreamSourceConduit channel) {
     private static final Logger log = Logger.getLogger(HttpClientConnection.class);
 
     private final Deque<HttpClientExchange> pendingQueue = new ArrayDeque<>();
-    private HttpClientExchange currentRequest;
-    private HttpResponseBuilder pendingResponse;
+    private volatile HttpClientExchange currentRequest;
+    private volatile HttpResponseBuilder pendingResponse;
```

Comment:
Why the volatile? This client is not designed to be thread safe, and should old be used from the IO thread that is tied to the connection.